src/pybind/mgr/dashboard/tests/__init__.py

```diff
@@ -38,16 +49,17 @@ def exec_dashboard_cmd(command_handler, cmd, **kwargs):
     inbuf = kwargs['inbuf'] if 'inbuf' in kwargs else None
     cmd_dict = {'prefix': 'dashboard {}'.format(cmd)}
     cmd_dict.update(kwargs)
-    if cmd_dict['prefix'] not in CLICommand.COMMANDS:
+
+    if command_handler:
         ret, out, err = command_handler(cmd_dict)
-        if ret < 0:
-            raise CmdException(ret, err)
-        try:
-            return json.loads(out)
-        except ValueError:
-            return out
-
-    ret, out, err = CLICommand.COMMANDS[cmd_dict['prefix']].call(mgr, cmd_dict, inbuf)
+    else:
+        dashboard_module = ModuleTestClass()
+        if cmd_dict['prefix'] in CLICommand.COMMANDS:
+            ret, out, err = CLICommand.COMMANDS[cmd_dict['prefix']].call(dashboard_module, cmd_dict,
+                                                                         inbuf)
+        else:
+            ret, out, err = dashboard_module.handle_command(inbuf, cmd_dict)
```

Comment:
Why is this needed?

Reply:
In order to test dashboard module ```handle_command``` that in turn calls ```create-self-signed-cert``` so we gain test coverage for both (previously ```handle_command``` was not being tested).
Here ```CLICommand.COMMANDS[cmd_dict['prefix']].call(dashboard_module,``` I replaced the mgr mock with dashboard module following the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

Reply:
Simplified the command testing through ```CLICommandTestMixin.exec_cmd```.